src/Meilisearch/Index.Documents.cs

```diff
@@ -430,6 +453,25 @@ public async Task<T> GetDocumentAsync<T>(int documentId, List<string> fields = d
                 .ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Delete documents from an index based on a filter.
+        /// </summary>
+        /// <remarks>Available ONLY with Meilisearch v1.2 and newer.</remarks>
+        /// <param name="query">A hash containing a filter that should match documents.</param>
+        /// <param name="cancellationToken">The cancellation token for this call.</param>
+        /// <returns>Return the task info.</returns>
+        public async Task<TaskInfo> DeleteDocumentsAsync(DeleteDocumentsQuery query,
```

Comment:
```suggestion
        public async Task<TaskInfo> DeleteDocumentsAsync(DocumentsQuery query,
```
You don't have to create a new class, you can use the existing `DocumentsQuery` WDYT?

Reply:
We can't here since `DocumentsQuery` contains more properties than `DeleteDocumentsQuery`